### PR TITLE
Perception point new

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17263

## Description
External PR of PerceptionPoint

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it) - https://github.com/demisto/etc/issues/17844
- [x] Code Review

### CODE REVIEW

**Comment by ShahafBenYakir** [comment]
@kirbles19 
[YML file](https://github.com/demisto/content/pull/3742/files#diff-9d90408bae1826267ee031ef79a55527)
[Description file](https://github.com/demisto/content/pull/3742/files#diff-5c4e5ef5b334ce96b7790a6ff8dfbcc4)

File: Integrations/PerceptionPoint/PerceptionPoint.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  defaultvalue: ""
+  type: 13
+  required: false
+- display: Trust any certificate (insecure)
+  name: insecure
+  defaultvalue: ""
+  type: 8 
+  required: false
+- display: Use system proxy
+  name: proxy
```
Comment (yuvalbenshalom) [inline Integrations/PerceptionPoint/PerceptionPoint.yml]: proxy paremeter is here, but there is no proxy handling in the code.
use `handle_proxy()`

File: Integrations/PerceptionPoint/PerceptionPoint.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        demisto.results(entry)
+    except Exception:
+        return_error(
+            'An error occurred while trying to release email. Please contact PerceptionPoint support')
+
+
+''' COMMAND CLASSIFIER'''
+try:
+    if demisto.command() == 'test-module':
+        list_url = build_request_url(LIST)
```
Comment (yuvalbenshalom) [inline Integrations/PerceptionPoint/PerceptionPoint.py]: please put this in a test function

File: Integrations/PerceptionPoint/PerceptionPoint.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        }
+        email_release_response = release_email_and_get_message(scan_id_to_release)
+        entry.update({'Contents': email_release_response,
+                      'ContentsFormat': formats['text'],
+                      'EntryContext': {'PP.Released': scan_id_to_release}}
+                     )
+        demisto.results(entry)
+    except Exception:
+        return_error(
+            'An error occurred while trying to release email. Please contact PerceptionPoint support')
```
Comment (yuvalbenshalom) [inline Integrations/PerceptionPoint/PerceptionPoint.py]: please add the str of the exception itself so you will have more info about what happened.

File: Integrations/PerceptionPoint/PerceptionPoint.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    except Exception:
+        return_error(
+            'An error occurred while trying to fetch new incidents. Please contact PerceptionPoint support')
+
+
+def release_email_and_get_message(scan_id_to_release):
+    try:
+        release_url = build_request_url(RELEASE).format(id_=scan_id_to_release)
+        _ = get_pp_api_result(url=release_url)
+        return 'Email with id {} was released Successfully!'.format(scan_id_to_release)
```
Comment (yuvalbenshalom) [inline Integrations/PerceptionPoint/PerceptionPoint.py]: please use py3 formatting style

File: Integrations/PerceptionPoint/PerceptionPoint.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        last_run_id = int(demisto.getLastRun().get('scan_id', 0))
+        req_args[API_CURSOR_ARG] = last_run_id
+        incidents_list = collect_incidents(**req_args)
+        report_incidents(incidents_list)
+        if incidents_list:
+            last_run_id = max(last_run_id, int(incidents_list[-1].get('Scan Id')))
+            demisto.setLastRun({'scan_id': int(last_run_id)})
+    except Exception:
+        return_error(
+            'An error occurred while trying to fetch new incidents. Please contact PerceptionPoint support')
```
Comment (yuvalbenshalom) [inline Integrations/PerceptionPoint/PerceptionPoint.py]: please add the str of the exception itself so you will have more info about what happened.

File: Integrations/PerceptionPoint/PerceptionPoint.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        return_error(err)
+
+
+def build_request_url(api_action):
+    return URL.format(endpoint=API_ACTIONS_DICT.get(api_action))
+
+
+def command_fetch_incidents():
+    try:
+        args_dict = demisto.args()
```
Comment (yuvalbenshalom) [inline Integrations/PerceptionPoint/PerceptionPoint.py]: please do not cascade arguments with kvargs.
extract demisto args explicitly in the beginning of _command function and pass them to the business object function explicitly.

**Comment by kirbles19** [comment]
@ShahafBenYakir All files reviewed.

File: Integrations/PerceptionPoint/PerceptionPoint.py [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        fetch_blocked = USER_PARAMS.get('fetch_blocked')
+        fetch_spam = USER_PARAMS.get('fetch_spam')
+        fetch_malicious = USER_PARAMS.get('fetch_malicious')
+        req_args = build_fetch_incident_types(fetch_blocked, fetch_malicious, fetch_spam)
+        last_run_id = int(demisto.getLastRun().get('scan_id', 0))
+        req_args[API_CURSOR_ARG] = last_run_id
+        incidents_list = collect_incidents(req_args)
+        report_incidents(incidents_list)
+        if incidents_list:
+            last_run_id = max(last_run_id, int(incidents_list[-1].get('Scan Id')))
```
Comment (yuvalbenshalom) [inline Integrations/PerceptionPoint/PerceptionPoint.py]: are ids sorted?

File: Integrations/PerceptionPoint/PerceptionPoint.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    return dict(fetch_type_dict)
+
+
+def create_incident(record):
+    record.pop('Attachment', None)
+    record['RawJSON'] = json.dumps(record)
+    return record
+
+
+def collect_incidents(params):
```
Comment (yuvalbenshalom) [inline Integrations/PerceptionPoint/PerceptionPoint.py]: still we have this functions that get "parmas" why??

**Pull request merged**